lib/pitchfork/http_server.rb

```diff
@@ -890,7 +890,7 @@ def process_client(client, worker, timeout_handler)
       env
     rescue => application_error
       handle_error(client, application_error)
-      env
+      env || @request.env
```

Comment:
```suggestion
      env || @request&.env
```

If `Pitchfork::HttpParser.new` raises [an `application_error`], better to fall back to a nil than an error